examples/safeExecute.js

```diff
@@ -0,0 +1,21 @@
+const safeExecute = async (retryLimit, work, prework, mutables) => {
+  let success = false;
+  let result = null;
+  let err = null;
+  for(let i = 0;i < retryLimit && !success; i++) {
+    try {
+      const preworkResults = await prework(mutables, err);
```

Comment:
We discussed that prework, which is a bad name of my own making, probably should be something else.  I think I last said savepoint, but then worried about the general DB use of that term.  The main point I wanted to get across is before a transaction can simply be retried, there might be application state that needs to be reset for the transaction retry to run sensibly.  It is not always the case state needs reset, only that if need be automatic retry mechanisms need a way to invoking it.  If the transaction is self contained and can run multiple times without causing harm, setting or using invalid state, then there is nothing extra to do.  How will this work if the user sets prework to null or undefined?

Reply:
In the current implementation it must be defined as a function. I can make it an optional argument tho.

As for the name prework, I'm fine to change it -- are you settled on savepoint or do you want to continue to brainstorm a name? We could call it 'prepwork'?